scipy/sparse/coo.py

```diff
@@ -137,16 +137,13 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                 except TypeError:
                     raise TypeError('invalid input format')
 
-                self.row = np.array(ij[0], copy=copy)
-                self.col = np.array(ij[1], copy=copy)
-                self.data = np.array(obj, copy=copy)
-
+                row, col = ij
```

Comment:
The replacement of the two `astype` calls with `asarray` calls does not prevent the creation of two arrays for each of `row` and `column` if `idx_dtype` is not the default `int` type.